pkg/controllers/csi/metadata/sqlite_gorm_client.go

```diff
@@ -0,0 +1,197 @@
+	if strings.Contains(path, "?") {
+		path += "&_foreign_keys=on"
+	} else {
+		path += "?_foreign_keys=on"
+	}
+
+	db, err := gorm.Open(sqlite.Open(path), &gorm.Config{Logger: logger.Default})
+
+	if err != nil {
+		return DBConn{}, err
+	}
+
+	return DBConn{db: db}, nil
+}
+
+// SchemaMigration runs gormigrate migrations to create tables
+func (conn *DBConn) SchemaMigration(_ context.Context) error {
+	m := gormigrate.New(conn.db, gormigrate.DefaultOptions, []*gormigrate.Migration{})
+	m.InitSchema(func(tx *gorm.DB) error {
+		err := tx.AutoMigrate(
+			&TenantConfig{},
+			&CodeModule{},
+			&OSMount{},
+			&AppMount{},
+			&VolumeMeta{},
+		)
+		if err != nil {
+			return err
+		}
+		// all other constraints, indexes, etc...
+		return nil
+	})
+
+	_ = m.Migrate()
+
+	return gormigrate.New(conn.db, gormigrate.DefaultOptions, []*gormigrate.Migration{
+		{
+			ID:      "202403041200",
+			Migrate: dataMigration,
+			Rollback: func(tx *gorm.DB) error {
+				return nil
+			},
+		},
+	}).Migrate()
+}
+
+func (conn *DBConn) CreateTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error {
+	return conn.db.WithContext(ctx).Create(tenantConfig).Error
+}
+
+func (conn *DBConn) UpdateTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error {
+	return conn.db.WithContext(ctx).Updates(tenantConfig).Error
+}
+
+func (conn *DBConn) DeleteTenantConfig(ctx context.Context, tenantConfig *TenantConfig) error {
+	return conn.db.WithContext(ctx).Delete(tenantConfig).Error
+}
+
+func (conn *DBConn) CreateCodeModule(ctx context.Context, codeModule *CodeModule) error {
+	return conn.db.WithContext(ctx).Create(codeModule).Error
+}
+
+func (conn *DBConn) ReadCodeModuleByVersion(ctx context.Context, version string) (*CodeModule, error) {
+	var codeModule CodeModule
+
+	result := conn.db.WithContext(ctx).First(&codeModule, "version = ?", version)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return &codeModule, nil
+}
+
+func (conn *DBConn) DeleteCodeModule(ctx context.Context, codeModule *CodeModule) error {
+	return conn.db.WithContext(ctx).Delete(codeModule).Error
+}
+
+func (conn *DBConn) ReadTenantConfigByTenantUUID(ctx context.Context, uid string) (*TenantConfig, error) {
+	var tenantConfig TenantConfig
+
+	result := conn.db.WithContext(ctx).First(&tenantConfig, "tenant_uuid = ?", uid)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return &tenantConfig, nil
+}
+
+func (conn *DBConn) CreateOSMount(ctx context.Context, osMount *OSMount) error {
+	return conn.db.WithContext(ctx).Create(osMount).Error
+}
+
+func (conn *DBConn) UpdateOSMount(ctx context.Context, osMount *OSMount) error {
+	return conn.db.WithContext(ctx).Updates(osMount).Error
+}
+
+func (conn *DBConn) ReadOSMountByTenantUUID(ctx context.Context, tenantUUID string) (*OSMount, error) {
+	var osMount OSMount
+
+	result := conn.db.WithContext(ctx).Preload("VolumeMeta").First(&osMount, "tenant_uuid = ?", tenantUUID)
```

Comment:
same here

Reply:
That won't work as stated above: in case of invalid tenantUUID it will create an invalid object instead of returning an error- at least that's what my tests showed.

Reply:
@wepudt what do you mean? for me works like a charm:

In case **Invalid** tenantUUID you get an error:
```
    sqlite_test.go:679: 
        	Error Trace:	/Users/andrii.soldatenko/work/dynatrace-operator/pkg/controllers/csi/metadata/sqlite_test.go:679
        	Error:      	Received unexpected error:
        	            	record not found
        	Test:       	TestSchemaMigration
--- FAIL: TestSchemaMigration (0.00s)

FAIL
```

My test:
```
func TestSchemaMigration(t *testing.T) {
	// new schema
	conn, err := NewDBAccess("file:csi_testdb?mode=memory")
	require.NoError(t, err)
	// new migrations
	err = conn.SchemaMigration(context.Background())
	require.NoError(t, err)

	tc := TenantConfig{
		Name:          "test",
		TenantUUID:    "test",
		ConfigDirPath: "/tmp",
	}
	require.NoError(t, conn.db.Create(&tc).Error)

	vm := VolumeMeta{
		ID:           "volume_id",
		PodUid:       "",
		PodName:      "php-pod",
		PodNamespace: "",
	}
	require.NoError(t, conn.db.Create(&vm).Error)

	om := OSMount{
		TenantUUID:      tc.TenantUUID,
		TenantConfigUID: tc.UID,
		VolumeMetaID:    vm.ID,
		MountAttempts:   10,
	}
	require.NoError(t, conn.db.Create(&om).Error)

	osMount := OSMount{TenantUUID: "test111"}
	require.NoError(t, conn.db.Preload("VolumeMeta").First(&osMount).Error)
}
```

Reply:
did it for all cases where it works (for the 
`TenantConfig` it does not work for `TenantUUID`) and adapted param checks to pass all tests (until refactoring to `sql.NullString`)